packages/ubuntu_desktop_installer/lib/pages/who_are_you/who_are_you_model.dart

```diff
@@ -94,21 +90,25 @@ class WhoAreYouModel extends ChangeNotifier {
 
   /// Whether the current input is valid.
   bool get isValid {
-    return realName.isNotEmpty &&
-        hostname.isNotEmpty &&
-        username.isNotEmpty &&
-        password.isNotEmpty &&
-        password == confirmedPassword &&
-        RegExp(kValidUsernamePattern).hasMatch(username) &&
-        RegExp(kValidHostnamePattern).hasMatch(hostname);
+    return _realName.value?.isValid == true &&
+        _hostname.value?.isValid == true &&
+        _username.value?.isValid == true &&
+        _password.value?.isValid == true &&
+        _confirmedPassword.value?.isValid == true;
+  }
+
+  void _initIdentity(ValueNotifier<ValidatedString?> identity, String? value) {
+    if (value != null) {
+      identity.value = ValidatedString(value);
+    }
   }
 
   /// Loads the identity data from the server, and resolves the system hostname.
   Future<void> loadIdentity() async {
     final identity = await _client.identity();
-    _realName.value = identity.realname?.orIfEmpty(null);
-    _hostname.value = identity.hostname?.orIfEmpty(null);
-    _username.value = identity.username?.orIfEmpty(null);
+    _initIdentity(_realName, identity.realname?.orIfEmpty(null));
+    _initIdentity(_hostname, identity.hostname?.orIfEmpty(null));
+    _initIdentity(_username, identity.username?.orIfEmpty(null));
```

Comment:
Instead of calling `orIfEmpty(null)` here, you could check whether the string is empty in `_initIdentity()`.